wait/member.go

```diff
@@ -570,3 +571,30 @@ func (a *MemberAwaitility) WaitForMemberStatus(criteria ...MemberStatusWaitCrite
 	})
 	return memberStatus, err
 }
+
+// WaitForMetricsService waits until there's a service called `member-operator-metrics` in the host
+// operator namespace
+func (a *MemberAwaitility) WaitForMetricsService() (corev1.Service, error) {
+	name := "member-operator-metrics"
+	var metricsSvc *corev1.Service
+	err := wait.Poll(a.RetryInterval, a.Timeout, func() (done bool, err error) {
+		metricsSvc = &corev1.Service{}
+		// retrieve the metrics service from the namespace
+		err = a.Client.Get(context.TODO(),
+			types.NamespacedName{
+				Namespace: a.Ns,
+				Name:      name,
+			},
+			metricsSvc)
+		if err != nil {
+			if errors.IsNotFound(err) {
+				a.T.Logf("Waiting for availability of service '%s' in namespace '%s'...\n", name, a.Ns)
+				return false, nil
+			}
+			return false, err
+		}
+		a.T.Logf("found service '%s'", metricsSvc.Name)
+		return true, nil
+	})
+	return *metricsSvc, err
+}
```

Comment:
Could be the content of the method moved to SingleAwaitility and parametrized by the name of the service? We could avoid code-duplication
```
func (a *SingleAwaitility) WaitForService(name string) (corev1.Service, error) {
```

Reply:
@MatousJobanek good point and it's actually what I've been doing in my current work for the performance measurement. If it's ok for you, this refactoring will be part of the upcoming PR (for the perfs)

Reply:
ok, if you are doing the change in the upcoming PR, then it's fine :wink:

Reply:
cool, thanks ;)